utils/shared.js

```diff
@@ -757,6 +757,18 @@ const isDateTimeFormat = (value) => {
     return typeof value == "string" && (/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(value));
 }
 
+const redactEmailLoginInfo = (loginDetails) => {
+    let amp = loginDetails.indexOf('@');    
+    for(let i = 0; i < amp; i++) {
+        if(i != 0 && i != 1 && i != amp - 1) {
+            loginDetails = loginDetails.substring(0, i) + "*" + loginDetails.substring(i + 1);
+        } 
+    }
+    return loginDetails
```

Comment:
Looks good, nice way to redact an email address!
Here's an example that also handles a very short email address and avoids rebuilding the string every iteration:
```javascript
const redactEmailLoginInfo = (loginDetails) => {
    const amp = loginDetails.indexOf('@');

    if (amp < 3) return loginDetails; // corner case (first two chars + last char before '@')
    
    let redacted = loginDetails.substring(0, 2); // Keep the first two characters
    
    for (let i = 2; i < amp - 1; i++) {
        redacted += '*';
    }

    redacted += loginDetails.charAt(amp - 1);
    redacted += loginDetails.substring(amp);
    
    return redacted;
};
```

Reply:
Thanks. The corner case is taken care in the original code. I leave it up to @anthonypetersen if he wants to change the function or not.

Reply:
https://github.com/episphere/connectFaas/pull/399/files#diff-23fab349744c189c5b655022c3edc921526e74945e34cdd65db82880911119d8R760